### PR TITLE
Feature/food description

### PR DESCRIPTION
**Pull request by mahendarsingh1**
added food list using dynamic data from json file and rendered it
Screenshot 2022-03-10 at 1 19 26 AM

### CODE REVIEW

File: src/components/foodDescription/FoodList.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import FoodListSection from "./FoodListSection";
+import './FoodList.css'
+
+
+function FoodList(props){
+
+    const { foodListData } = props;
+
+    let keys = Object.keys(foodListData);
```
Comment (jamruthsai) [inline src/components/foodDescription/FoodList.js]: Use const

File: src/components/foodDescription/FoodListSection.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+function FoodListSection(props){
+
+    const { items, category } = props;
+
+    return (
+        <>
+            <h2 id={category}>{category}</h2>
+            <p className="op-9">{items.length} ITEMS</p>
+
+            {items.map(item => {
```
Comment (jamruthsai) [inline src/components/foodDescription/FoodListSection.js]: Use separate fn

File: src/components/foodDescription/Overview.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import './Overview.css'
+
+function Overview(props){
+
+    const { categories } = props;
+        
+    return(
+        <div className="overview">
+            {
```
Comment (jamruthsai) [inline src/components/foodDescription/Overview.js]: Use separate fn

File: public/data/foodList.json [hunk 4]
```diff
@@ -0,0 +1,140 @@
+{
+    "Starters": [
+        {
+            "id":1,
+            "heading":"Chilli Paratha",
```
Comment (kprakhar04) [inline public/data/foodList.json]: heading is not a good key for this, you can use **name**

File: src/components/foodDescription/FoodDescription.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    
+    constructor(props){
+        super(props);
+        this.state = {
+            foodListData: {},
+        }
+    }
+
+    componentDidMount(){
+        fetch('./data/foodList.json')
```
Comment (kprakhar04) [inline src/components/foodDescription/FoodDescription.js]: use some ApiUtility for fetch, don't call it from here directly

File: src/components/foodDescription/FoodDescription.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+            .then(res => res.json())
+            .then(data => this.setState({foodListData : data}))
+            .catch(err=> console.log(err));
+    }
+
+    render(){
+        return (
+            <div className="food-description flex">
+                <Overview categories={Object.keys(this.state.foodListData)} />
```
Comment (kprakhar04) [inline src/components/foodDescription/FoodDescription.js]: use destructuring

File: src/components/foodDescription/FoodList.css [hunk 7]
```diff
@@ -0,0 +1,13 @@
+.food-list{
+  width: 600px;
+  height: 620px;
+  overflow-y: auto;
+}
+.food-list>h2{
```
Comment (kprakhar04) [inline src/components/foodDescription/FoodList.css]: use rem

File: src/components/foodDescription/FoodList.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    let listSection = keys.map(category=>{
```
Comment (kprakhar04) [inline src/components/foodDescription/FoodList.js]: use some helper function for rendering logic plus use const

File: src/components/foodDescription/FoodListItem.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+function FoodListItem(props) {
+
+    const { item } = props;
+    const { heading, amount, description, type } = item;
+
+    return (
+        <>
+            <div className="food-list-item">
+                <img src={img} className={`margin-tb-8 h-15 ${type}` } alt=""/>
```
Comment (kprakhar04) [inline src/components/foodDescription/FoodListItem.js]: alt data missing

File: src/components/foodDescription/FoodListItem.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+                <div className="flex-space-between w-90">
+                    <div>
+                        <h3 className="margin-tb-8">{heading}</h3>
+                        <p className="margin-tb-8">₹{amount}</p>
+                    </div>
+                    <button className="food-list-item-button">ADD</button>
```
Comment (kprakhar04) [inline src/components/foodDescription/FoodListItem.js]: use text-transform

File: src/components/foodDescription/FoodListItem.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    const { heading, amount, description, type } = item;
+
+    return (
+        <>
+            <div className="food-list-item">
+                <img src={img} className={`margin-tb-8 h-15 ${type}` } alt=""/>
+                <div className="flex-space-between w-90">
+                    <div>
+                        <h3 className="margin-tb-8">{heading}</h3>
+                        <p className="margin-tb-8">₹{amount}</p>
```
Comment (kprakhar04) [inline src/components/foodDescription/FoodListItem.js]: don't hardcode rupee symbol

File: src/components/foodDescription/FoodDescription.js (hunk 6 above)
Comment (kprakhar04) [inline src/components/foodDescription/FoodDescription.js]: Name this component Sidebar or something similar and try to make it a reusable one

**Pull request closed without merging**